### DISCUSSION TITLE
Gapless looping of compressed files?

### DISCUSSION THREAD

**Discussion by jbmaxwell** [sbooth/SFBAudioEngine#191] [Q&A] [answered]
Just curious if this library supports gapless looping of compressed audio (e.g., AAC)? 
I know AVAudioPlayer is supposed to, but I'm finding it generally doesn't loop without gaps.

**Comment by sbooth** [comment, marked as answer]
Generally yes, although it depends on whether the decoder for the specific format knows and decodes the correct number of PCM samples (meaning it is able to discard priming, remainder, padding, etc). This should work for MP3 files with a Xing header. It should also work for AAC using the Core Audio decoder (at least for me on macOS 11.2.3). I *think* it should work for Vorbis and Musepack too although I haven't tested it.

Are you primarily interested in AAC?

**Reply by jbmaxwell** [reply]
Excellent, thanks for the quick reply. Yes, our files are all AAC, at this point.

**Reply by sbooth** [reply]
In case the classes are not documented well the way to accomplish this is using [SFBLoopableRegionDecoder](https://github.com/sbooth/SFBAudioEngine/blob/master/Decoders/SFBLoopableRegionDecoder.h) with a repeat count.

**Reply by jbmaxwell** [reply]
Great, thanks for the tip!
